### PR TITLE
Add Undo AI edit per-row affordance in dictation history

### PR DESCRIPTION
**Pull request by moona3k**
## What

Per-row **Undo AI edit** menu on dictation history — flip any AI-cleaned row back to the raw STT output. Reversible, persistent, plays nicely with the existing Copied badge slot.

```
┌──────────────────────────────────────────────────────┐
│ 9:33 AM · 0:08 · 🎤            [▶]  [⧉]  [⋯]         │
│ Hey, are we still on for coffee?                     │
└──────────────────────────────────────────────────────┘
                                            │
                                            ▼ click ⋯
                                ┌──────────────────────┐
                                │  Export Audio        │
                                │  Undo AI edit        │
                                │  ──────────────────  │
                                │  Delete              │
                                └──────────────────────┘
                                            │
                                            ▼ click
┌──────────────────────────────────────────────────────┐
│ 9:33 AM · 0:08 · 🎤  ⟨Raw⟩     [▶]  [⧉]  [⋯]         │
│ hey so are we still on for coffee, i think we shou…  │
└──────────────────────────────────────────────────────┘
                                            │
                                            ▼ click ⋯
                                ┌──────────────────────┐
                                │  Export Audio        │
                                │  Re-apply AI edit    │
                                │  ──────────────────  │
                                │  Delete              │
                                └──────────────────────┘
```

## How it works

\`Dictation\` already stored both \`rawTranscript\` and \`cleanTranscript\` separately. This PR adds a \`displayRawTranscript: Bool\` column and a \`Dictation.displayText\` computed property, then routes **every** display consumer — history list, copy-to-clipboard, menu-bar Recent Dictations / Paste Last, CLI list+search — through it. The toggle persists across launches and round-trips cleanly.

The naïve approach in the original task brief was to flip \`processingMode\` from \`.clean\` to \`.raw\` on the saved row. A grep of every display consumer showed they all use the pattern \`cleanTranscript ?? rawTranscript\` — nobody reads \`processingMode\`. Flipping it would silently no-op. A dedicated override column is the smallest correct change.

## How to verify

1. Find a dictation in history with an AI-cleaned transcript.
2. Click the row overflow menu (⋯) → **Undo AI edit**.
3. Row now shows the raw transcript + a muted **Raw** capsule; copy and export use the raw text.
4. Same menu now reads **Re-apply AI edit** — round-trips back cleanly.
5. Rows with no AI edit (or where \`cleanTranscript == rawTranscript\`) don't show the toggle.

## Polish + bot review (all addressed)

- Subtle scale+opacity spring on the **Raw** badge entry/exit
- Transcript text crossfades via \`.contentTransition(.opacity)\` on toggle
- In-place row mutation instead of a full re-fetch (avoids flicker on the active row)
- Mock repository now mirrors prod's no-op short-circuit semantics

## Tests

\`swift test\` — **2508 XCTest + 16 Swift Testing pass** (baseline + 13 new). Repository round-trip, ViewModel toggle/reversibility, model decoding without the new column key, edge cases (unknown id, no AI edit).

## Notes

- Migration is forward-only and re-run safe (pre-checks column existence).
- Live capture-time paste still uses \`cleanTranscript ?? rawTranscript\` — this PR is purely post-hoc history revert.
- No telemetry event: adding a \`TelemetryEventName\` case is a two-repo change with the website Worker allowlist; leaving as a follow-up if usage signal is wanted.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Per-dictation "Undo AI edit" toggle to switch between raw and AI-edited transcript display.
  * Toggle updates History rows, playback bar, recent-items menu preview, paste actions, CLI output, and clipboard contents.
  * "Raw" badge appears when raw transcript is active; menu action flips between “Undo AI edit” and “Re‑apply AI edit”.

* **Chores / Reliability**
  * Safe database migration added to persist the per-dictation setting.

* **Tests**
  * Added unit tests for toggle behavior, persistence, reversibility, no-op cases, and legacy decoding.

[Review Change Stack](https://app.coderabbit.ai/change-stack/moona3k/macparakeet/pull/277)

### CODE REVIEW

File: Sources/MacParakeet/Views/History/DictationHistoryView.swift [hunk 1]
```diff
@@ -310,8 +329,11 @@ struct DictationCardRow: View {
 
                     CardMenuButton(
                         hasAudio: dictation.audioPath != nil,
+                        hasAIEdit: dictation.hasAIEdit,
+                        isShowingRaw: dictation.displayRawTranscript,
                         onDownloadAudio: { onDownloadAudio?() },
-                        onDelete: { onDelete() }
+                        onDelete: { onDelete() },
+                        onToggleAIEdit: { onToggleAIEdit?() }
                     )
```
[the remark this answers is not in the document]
Comment (moona3k) [reply Sources/MacParakeet/Views/History/DictationHistoryView.swift:337]: Addressed in 0132d8f1. The call site now gates hasAIEdit on onToggleAIEdit != nil, and CardMenuButton keeps the callback optional and only adds the AI-edit menu item when the callback exists. Full swift test passed locally.

File: Tests/MacParakeetTests/ViewModels/ViewModelMocks.swift [hunk 2]
```diff
@@ -76,6 +76,16 @@ final class MockDictationRepository: DictationRepositoryProtocol, @unchecked Sen
...
     }
 
+    var setDisplayRawTranscriptCalls: [(id: UUID, value: Bool)] = []
+    @discardableResult
+    func setDisplayRawTranscript(id: UUID, value: Bool) throws -> Bool {
+        setDisplayRawTranscriptCalls.append((id, value))
+        guard let idx = dictations.firstIndex(where: { $0.id == id }) else { return false }
+        dictations[idx].displayRawTranscript = value
+        dictations[idx].updatedAt = Date()
+        return true
```
[the remark this answers is not in the document]
Comment (moona3k) [reply Tests/MacParakeetTests/ViewModels/ViewModelMocks.swift]: Verified and addressed. MockDictationRepository now mirrors production no-op semantics by returning true without mutating displayRawTranscript or updatedAt when the value is unchanged. I also added repository assertions that same-value toggles do not bump updatedAt. Focused tests and full swift test passed locally.

File: Sources/MacParakeetViewModels/DictationHistoryViewModel.swift [hunk 3]
```diff
@@ -223,6 +226,31 @@ public final class DictationHistoryViewModel {
...
+            // events). Keeping this PR scoped to Undo AI edit — telemetry can
+            // be added as a follow-up if usage signal is needed.
+        } catch {
+            logger.error("Failed to toggle displayRawTranscript for \(dictation.id): \(error.localizedDescription)")
+            return
+        }
+        // Reload without bouncing stats — the toggle leaves duration/wordCount
+        // untouched, so the lifetime/daily counters can't have shifted.
+        loadDictations(shouldRefreshStats: false)
+    }
```
[the remark this answers is not in the document]
Comment (moona3k) [reply Sources/MacParakeetViewModels/DictationHistoryViewModel.swift:270]: Verified current code. The view model now persists the toggle and updates the matching row in groupedDictations in place via updateDictationInPlace, so this no longer reloads and re-groups the full history list. Covered by DictationHistoryViewModelTests; full swift test passed locally.

**Comment by moona3k** [comment]
Review follow-up pushed in 0132d8f1.

- Verified Gemini suggestion: toggle persistence now updates groupedDictations in place instead of reloading the full history list.
- Verified CodeRabbit mock semantics: MockDictationRepository now preserves production no-op updatedAt behavior; added explicit no-op timestamp assertions.
- Addressed CodeRabbit menu callback guard: the AI-edit menu item now requires both hasAIEdit and a real onToggleAIEdit callback.

Validation: swift test passed locally (2508 XCTest tests, 10 skipped hardware/calendar-gated tests, plus 16 Swift Testing tests). Inline review threads replied to and resolved.
